src/invision/components/movie-item/movie-item.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './movie-item.scss';
+import MoviePopup from '../movie-popup/movie.popup';
+import DeletePopup from '../delete-movie/delete-popup';
+
+
+export default class MovieItem extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isEditMovieVisible: false,
+      isMenuVisible: false,
+      isDeletePopup: false,
+    }
+    this.showMenu = () => {
+      this.setState({ isMenuVisible: 'block' });
+    }
+
+    this.hideMenu = () => {
+      this.setState({ isMenuVisible: 'none' });
+    }
+
+    this.onModalClose = this.onModalClose.bind(this);
+    this.onDeletePopupClose = this.onDeletePopupClose.bind(this);
+  }
+
+  editMovie() {
+    this.setState({ isEditMovieVisible: true });
+  }
+
+  onModalClose(visible) {
+    this.setState({ isEditMovieVisible: visible })
+  }
+  deletePopup(){
+    this.setState({ isDeletePopup: true })
+  }
+
+  onDeletePopupClose(isDelete) {
+    if (isDelete) {
+      // delete move code
+    } else {
+      this.setState({ isDeletePopup: false });
+    }
+  }
+
+  render() {
+    return (
+      <div className="item movie-list-item">
+        <MoviePopup visible={this.state.isEditMovieVisible} title="Edit Movie" movie={this.props.movie} onModalClose={this.onModalClose} />
```

Comment:
If you do not need something - do not add it to the DOM. Also, you have a side effect: when I open Edit, change something, click outside, then open Edit again - I see all my previous changes